modules/builds-running-entitled-builds-with-sharedsecret-objects.adoc

```diff
@@ -0,0 +1,100 @@
+
+Under the [examples directory](examples) there is both a simple example of using a `SharedConfigMap` in a `Pod`, and an example of performing an OpenShift `Build` from a `BuildConfig` that uses Red Hat Entitlements available from your Red Hat Subscription when your subscription credentials are stored on your cluster via OpenShift Insights Operator, and resulting `Secret` for those credentials is made available to additional `Namespaces` via a `SharedSecret`.  Instructions for the `Pod` example are [here](docs/simple-example.md), and for the `BuildConfig` example are [here](docs/build-with-rhel-entitlements.md).
+
+# BuildConfig example
+
+From the root directory, create/apply the various API objects defined in the YAML files in the `./examples` directory.
+There are two examples there, where both use [the same namespace](../examples/00-namespace.yaml).
+
+This document describes the [OpenShift Builds with Red Hat Entitlements](../examples/build-with-rhel-entitlements) example.
+
+The example is an OpenShift `BuildConfig` that utilizes a `SharedSecret` for the `etc-pki-entitlement` secret in the
+`openshift-config-managed` namespace, which is set up by the OpenShift Insights Operator if your install pull secret in fact has a valid
+Red Hat subscription that entitles you to install Red Hat RPMs.  With a reference to this `SharedSecret`, the `BuildConfig`
+is able to use your subscription information to install entitled RPMs when building an image.  Separate `Role` and
+`RoleBinding` objects are defined for the `BuildConfig` and its use of the `SharedSecret`.
+
+To install this example, run:
+
+```shell
+$ kubectl apply -R -f ./examples
+```
+
+You can start a `Build` for this `BuildConfig`, and follow the logs, via
+
+```shell
+$ oc start-build my-csi-bc -F
+```
+
+and watch as Red Hat entitled content via your subscription is installed during the running of the Docker Strategy build.
+See the next section first however, if you have never used OpenShift Insights Operator to inject your subscription
+credentials into your cluster.
+
+## Notes or setting up your cluster so that Insights Operator injects your subscription credentials for use in your cluster
+
+Tech Preview must be enabled on your OCP cluster to use the Shared Resources CSI Driver and to sync the Simple Content
+Access certificate of your Red Hat account.
+See [this link](https://docs.openshift.com/container-platform/4.9/post_installation_configuration/cluster-tasks.html#nodes-cluster-enabling-features-cli_post-install-cluster-tasks)
+if you want to learn how to convert your cluster to a Tech Preview cluster.
+
+Once that is sorted out, as noted up top, the Insights Operator will store your subscription credentials in the `Secret`
+named `etc-pki-entitlement` in the `openshift-config-managed` namespace.  However, you'll need to perform some management
+actions around your subscription and your cluster to get the Insights Operator to actually create and maintain that `Secret`
```

Comment:
Perhaps we could say what actions? Or at least link some other documentation?

Reply:
@gabemontero Should we make note of this in the 4.10 docs?